lib/compress/zstd_compress.c

```diff
@@ -1651,7 +1672,7 @@ ZSTD_resetCCtx_byAttachingCDict(ZSTD_CCtx* cctx,
         params.cParams = ZSTD_adjustCParams_internal(*cdict_cParams, pledgedSrcSize, 0);
         params.cParams.windowLog = windowLog;
         ZSTD_resetCCtx_internal(cctx, params, pledgedSrcSize,
-                                ZSTDcrp_continue, zbuff);
+                                ZSTDcrp_forceMemset, zbuff);
```

Comment:
I'm not sure to follow the objective of this change.
Is that for tests ?

Reply:
There are three cases for this parameter, which is used to control the behavior of ZSTD_reset_matchState():
- Don't clear the table (regardless of compression strategy)
- Clear the table (regardless of compression strategy)
- determine whether to clear the table based on the compression strategy (i.e. whether it is zstd_fast etc...)